core/sail/nativerdf/src/test/java/org/eclipse/rdf4j/sail/nativerdf/MemoryOverflowToDiskTest.java

```diff
@@ -0,0 +1,51 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Eclipse RDF4J contributors.
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Distribution License v1.0
+ *  which accompanies this distribution, and is available at
+ *  http://www.eclipse.org/org/documents/edl-v10.php.
+ ******************************************************************************/
+
+package org.eclipse.rdf4j.sail.nativerdf;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
+import org.eclipse.rdf4j.model.Model;
+import org.eclipse.rdf4j.sail.base.SailStore;
+import org.junit.jupiter.api.Test;
+
+public class MemoryOverflowToDiskTest {
+
+	@Test
+	public void testCleaner() throws IOException, InterruptedException {
+		File file = Files.createTempDirectory("model").toFile();
+
+		Model model = createModel(file);
+		assertTrue(file.exists());
+		model = null;
+
+		System.gc();
+		while (file.exists()) {
+			System.gc();
+			Thread.sleep(10);
+			System.out.println();
+		}
+
+		assertFalse(file.exists());
```

Comment:
Doesn't this run a risk of looping endlessly? Not sure we have a global test timeout configured here.

Reply:
True. I'll fix it.

Reply:
Fixed it with this: https://mkyong.com/junit5/junit-5-timeouts-examples/